includes/Core/Tags/Google_Tag_Gateway/Google_Tag_Gateway_Settings.php

```diff
@@ -27,6 +27,53 @@ class Google_Tag_Gateway_Settings extends Setting {
 	 */
 	const OPTION = 'googlesitekit_google_tag_gateway';
 
+	/**
+	 * Registers the GTG settings.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		parent::register();
+
+		$this->add_option_default_filters();
+	}
+
+	/**
+	 * Adds default filters to the GTG settings to ensure default values are present in the saved option.
+	 *
+	 * This is particularly important for newly added settings like `isGTGDefault` which won't exist
+	 * in the database for existing sites. The filters ensure that:
+	 * 1. If no GTG settings exist at all, the complete default array is returned
```

Comment:
This is incorrect. If the settings are not saved yet, the registered default is already returned without this (and passes through a different filter: `default_option_*`). This is a safeguard against a non-array being saved as the value, which probably isn't really needed. It's from very old code but doesn't hurt either I guess.